pkg/registry/core/service/storage/storage.go

```diff
@@ -625,6 +625,12 @@ func patchAllocatedValues(after After, before Before) {
 		for i := range newSvc.Spec.Ports {
 			p := &newSvc.Spec.Ports[i]
 			if p.NodePort == 0 {
+				// skip setting NodePort in case AllocateLoadBalancerNodePorts is false
+				if newSvc.Spec.Type == api.ServiceTypeLoadBalancer &&
+					newSvc.Spec.AllocateLoadBalancerNodePorts != nil &&
+					*newSvc.Spec.AllocateLoadBalancerNodePorts == false {
+					continue
+				}
```

Comment:
```suggestion
					!*newSvc.Spec.AllocateLoadBalancerNodePorts {
					continue
				}
```

Reply:
we can't use the *AllocateLoadBalancerNodePorts directly as the value can be null and bring nil pointer panics. especially while upgrading control plane without updating the service specs.

Reply:
I mean we just need to use `!*newSvc.Spec.AllocateLoadBalancerNodePorts` instead of `*newSvc.Spec.AllocateLoadBalancerNodePorts == false`

Reply:
updated.